src/Microsoft.AspNet.Routing/Template/TemplateRoute.cs

```diff
@@ -202,26 +202,20 @@ public virtual string GetVirtualPath(VirtualPathContext context)
 
             // Validate that the target can accept these values.
             var childContext = CreateChildVirtualPathContext(context, values.AcceptedValues);
+
             var path = _target.GetVirtualPath(childContext);
             if (path != null)
             {
                 // If the target generates a value then that can short circuit.
-                context.IsBound = true;
```

Comment:
not sure if I understand this change, if a target generates a value why do we not mark context.IsBound?